### PR TITLE
TransportReplicationAction subclasses shouldn't have to resolve shards

### PR DESCRIPTION
**Pull request by bleskes**
TRA currently resolves incoming requests to IndexShards in order to acquire operations locks on them. There is no need for all subclasses to have to go through the same IndicesService/IndexService song and dance. Also, doing it once means we don't need to worry about edge cases where the shard is removed while a TRA is in flight.

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java [hunk 1]
```diff
@@ -550,6 +562,12 @@ public void onFailure(Exception e) {
         }
     }
 
+    /** resolves a shardId to an IndexShard. protected for overriding in tests **/
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: This comment ("`protected`") is not consistent with the method access modifier `private`.

**Review by jasontedor — CHANGES_REQUESTED** [review]
Good change, I left one comment.

**Comment by bleskes** [comment]
thx @jasontedor . I merged from master and pushed a new commit.

**Review by jasontedor — APPROVED** [review]
LGTM, as long as the CI is happy (I don't see any reason for it not to be).

**Pull request merged**